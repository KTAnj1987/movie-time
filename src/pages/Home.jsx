import React, {useEffect, useState} from "react";
import { getBanners } from "../api";

const Home = () => {
  const [images, setImages] = useState([])

  useEffect(() => {
    getBanners().then(data=>{
      setImages(data.object.bannerHomepages)
    })
  }, []);
  return (
    <>
      <h2>Find Your Favorite Movies with Movie Time: A Free search Website</h2>
      <p>
        If you are into details of movies, Movie Time is the website for you.
        This website offers a huge library of movie content.
      </p>
      <div style={{
        width: '600px',
        height: '600px',
        aspectRatio: 'auto 600 / 600'
      }} >
      {images && images.map((image,index) => (
        <img style={{width: '100%', height: 'auto'}} key={`image-${index}`} src={`https://static-qa.edclinic.co.uk/cdn-cgi/image/width=471,quality=93,f=auto/${image.imageLinkDesktop}`}/>
      ))}
      </div>
      
      
    </>
  );
};

export default Home;
