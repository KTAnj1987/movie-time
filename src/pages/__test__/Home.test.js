import React from "react";
import renderer from "react-test-renderer";
import Home from "../Home";

jest.mock("../../api", () => ({
  getBanners: async() => ({object:{bannerHomepages:[]}}),
}));

describe("should render Home component", () => {
  it("shold match home page snapshot", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
