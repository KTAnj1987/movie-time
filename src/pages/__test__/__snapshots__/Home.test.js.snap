// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home component shold match home page snapshot 1`] = `
[
  <h2>
    Find Your Favorite Movies with Movie Time: A Free search Website
  </h2>,
  <p>
    If you are into details of movies, Movie Time is the website for you. This website offers a huge library of movie content.
  </p>,
  <div
    style={
      {
        "height": "auto",
        "width": "100%",
      }
    }
  />,
]
`;
